pom.xml

```diff
@@ -111,12 +137,68 @@
                 <cmake.cflags>-m32</cmake.cflags>
             </properties>
         </profile>
+        <!-- native compilation -->
+        <profile>
+            <id>build-shared-lib</id>
+            <activation>
+                <property>
+                    <name>!shared-lib.skip</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <!-- cmake configure and build -->
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>exec-maven-plugin</artifactId>
+                        <version>1.3.2</version>
+                        <executions>
+                            <!-- cmake configure -->
+                            <execution>
+                                <id>cmake-configure</id>
+                                <phase>generate-sources</phase>
+                                <goals>
+                                    <goal>exec</goal>
+                                </goals>
+                                <configuration>
+                                    <executable>${cmake.command}</executable>
+                                    <workingDirectory>mvn-build</workingDirectory>
+                                    <arguments>
+                                        <argument>-G${cmake.generator}</argument>
```

Comment:
where's CMAKE_PREFIX_PATH?

Reply:
Isn't necessary, because the deps build takes care of setting up CMAKE_MODULE_PATH, after which we just load the config scripts and we're good to go. An external user should not have to provide CMAKE_PREFIX_PATH unless they did their own compiling and are not automatically building deps.